common/src/main/java/com/google/auto/common/BasicAnnotationProcessor.java

```diff
@@ -505,74 +525,402 @@ private ImmutableSetMultimap<Class<? extends Annotation>, Element> toClassKeyedM
+   * information about the type. For example, if the information about the type is originating from
+   * a source file, the elements will be returned in source code order. (However, in that case the
+   * ordering of implicitly declared elements, such as default constructors, is not specified.)"
    *
-   * <p>Packages declared (and annotated) in {@code package-info.java} are tracked as deferred
-   * packages, type elements are tracked directly, and all other elements are tracked via their
-   * nearest enclosing type.
+   * <p>Simple name is saved since comparing the toString as a subtitle is not reliable when at
+   * least one parameter references ERROR, possibly because it is not generated yet. For example,
+   * method {@code void m(SomeGeneratedClass sgc)}, before the generation of {@code
+   * SomeGeneratedClass} has the toString {@code m(SomeGeneratedClass)}; however, after the
+   * generation it will have toString equal to {@code m(test.SomeGeneratedClass)} assuming that the
+   * package name is "test".
    */
-  private static final class ElementName {
-    private enum Kind {
-      PACKAGE_NAME,
-      TYPE_NAME,
+  private static final class ExecutableElementBluePrint extends ElementBluePrint {
+    private final TypeElementBluePrint enclosingTypeElementBluePrint;
+    private final Name simpleName;
+    private final int ordinalOverloadPosition;
+
+    private ExecutableElementBluePrint(Element element) {
+      super(element);
+      TypeElement enclosingTypeElement = getEnclosingType(element);
+      this.enclosingTypeElementBluePrint = new TypeElementBluePrint(enclosingTypeElement);
+      this.simpleName = element.getSimpleName();
+      int ordinalOverloadPosition = 1;
+      Iterator<? extends Element> iter = enclosingTypeElement.getEnclosedElements().iterator();
+      Element elt = iter.next(); // The collection is not empty
+      while (!element.equals(elt)) {
+        if (elt.getSimpleName().equals(element.getSimpleName()) && isAcceptableElementKind(elt)) {
+          ordinalOverloadPosition++;
+        }
+        elt = iter.next();
+      }
+      this.ordinalOverloadPosition = ordinalOverloadPosition;
     }
 
-    private final Kind kind;
-    private final String name;
+    @Override
+    protected boolean isAcceptableElementKind(Element element) {
+      return element.getKind() == CONSTRUCTOR || element.getKind() == METHOD;
+    }
 
-    private ElementName(Kind kind, Name name) {
-      this.kind = checkNotNull(kind);
-      this.name = name.toString();
+    @Override
+    Optional<? extends ExecutableElement> getElement(Elements elementUtils) {
+      return enclosingTypeElementBluePrint
+          .getElement(elementUtils)
+          .map(
+              typeElement -> {
+                ListIterator<? extends Element> listIter =
+                    typeElement.getEnclosedElements().listIterator();
+                for (int i = 0; i < ordinalOverloadPosition; i++) { // ordinalOverloadPosition >= 1
```

Comment:
Even though it's a bit less efficient, I think it would be worth using a stream to get a list of elements where `elt.getSimpleName().equals(simpleName) && isAcceptableElementKind(elt)`. Then I think you're just looking for `list.get(ordinalOverloadPosition)`, or maybe ±1.

Reply:
I have changed the implementation as requested.